defrag/modules/docs.py

```diff
@@ -0,0 +1,108 @@
+from defrag import app
+from pydantic import BaseModel
+
+
+class LunrDoc(BaseModel):
+    number: str
+    contents: str
+    permalink: str
+    title: str
+
+
+indexes = {
+    "leap": {
+        "url": "https://doc.opensuse.org/documentation/leap/reference/single-html/book-reference/index.html",
+        "index": None,
+    },
+    "tumbleweed": {
+        "url": "https://raw.githubusercontent.com/openSUSE/openSUSE-docs-revamped-temp/gh-pages/search/search_index.json",
+        "index": None
+    }
+}
+
+
+async def get_data(source: str) -> Any:
+    async with Req(indexes[source]["url"]) as result:
+        if source == "tumbleweed":
+            return await result.read()
+        return await result.text()
+
+
+def make_soup(data_str: str) -> List[LunrDoc]:
+    root = BeautifulSoup(data_str, 'html.parser')
+    chapters = root.find_all(class_="chapter")
+    res = []
+    for c in chapters:
+        titles = c.find_all(class_="titlepage")
+        contents = [c.get_text() for c in c.find_all("p")]
+        for i, t in enumerate(titles):
+            names = t.find_all(class_="name")[0].get_text()
+            permalink = t.find_all(class_="permalink")[0]["href"]
+            number = t.find_all(class_="number")[0].get_text().strip()
+            res.append(LunrDoc(number=number, title=names,
+                       permalink=permalink, contents="".join(contents[i])))
+    return res
+
+
+def setIndex(docs: List[LunrDoc]) -> Index:
+    listed = [doc.dict() for doc in docs]
+    idx = lunr(ref="permalink", fields=("contents", "title"), documents=listed)
+    indexes["leap"]["index"] = idx
+    return idx
+
+
+def thin_prime(item: Dict[str, Any], source: str) -> Dict[str, Any]:
+    built = {k: item[k] for k in item if k in ["score", "ref"]}
+    built["source"] = source
+    return built
+
+
+def sorted_on_score(items: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
+    return sorted(items, key=lambda x: x["score"], reverse=True)
+
+
+def render(items: List[Dict[str, Any]], source: str) -> List[Dict[str, Any]]:
+    return sorted_on_score((list(map(lambda item: thin_prime(item, source), items))))
+
+
+async def search(keywords: str, source: str) -> List[Dict[str, Any]]:
+    global indexes
+    if source == "leap":
+        idx = indexes[source]["index"]
+        if not idx:
+            idx = setIndex(make_soup(await get_data(source)))
+            indexes[source]["index"] = idx
+        return idx.search(keywords)
+    elif source == "tumbleweed":
+        idx = indexes[source]["index"]
+        if not idx:
+            raw = await get_data(source)
+            serialized: Dict[str, Any] = json.loads(raw)["index"]
+            new_idx = Index.load(serialized)
+            indexes[source]["index"] = new_idx
+            idx = new_idx
+        return idx.search(keywords)
+    else:
+        raise Exception(
+            f"Unable to work with source {source} and keywords {keywords}")
+
+
+@app.get("/single_search/{source}/{keywords}")
```

Comment:
I think the URL could be better:
```suggestion
@app.get("/" + __MOD_NAME__+ "/single/{source}/{keywords}")
```